File: app/src/main/java/chess/android/arduino/telescope/coordinates/Declination.java
package chess.android.arduino.telescope.coordinates;

public class Declination extends Coordinate{

    public Declination(int degree, int minute, int second) {
        super(0, minute, second);
        this.setDegree(degree);
    }

    @Override
    public void plusMinute() {
        super.plusMinute();
        if (this.getMinute()==0) {
            this.plusDegree();
        }
    }

    @Override
    public void minusMinute() {
        super.minusMinute();
        if (this.getMinute()==59) {
            this.minusDegree();
        }
    }

    @Override
    public double toDegrees() {
        return ((this.getDegree()*3600.0d)+(this.getMinute()*60.0d)+this.getSecond())/3600.0d;
    }
}

File: app/src/main/java/chess/android/arduino/telescope/coordinates/Declination.kt
package chess.android.arduino.telescope.coordinates

class Declination(degree: Int, minute: Int, second: Int) : Coordinate(0, minute, second) {
    override fun plusMinute() {
        super.plusMinute()
        if (minute == 0) {
            plusDegree()
        }
    }

    override fun minusMinute() {
        super.minusMinute()
        if (minute == 59) {
            minusDegree()
        }
    }

    override fun toDegrees(): Double {
        return (degree * 3600.0 + minute * 60.0 + second) / 3600.0
    }

    init {
        this.degree = degree
    }
}